src/libsync/clientsideencryption.cpp

```diff
@@ -1207,38 +1296,61 @@ void ClientSideEncryption::generateKeyPair(const AccountPtr &account)
 
     if(EVP_PKEY_keygen_init(ctx) <= 0) {
         qCInfo(lcCse()) << "Couldn't initialize the key generator";
+        failedToInitialize(account);
         return;
     }
 
     if(EVP_PKEY_CTX_set_rsa_keygen_bits(ctx, rsaKeyLen) <= 0) {
         qCInfo(lcCse()) << "Couldn't initialize the key generator bits";
+        failedToInitialize(account);
         return;
     }
 
     auto localKeyPair = PKey::generate(ctx);
     if(!localKeyPair) {
         qCInfo(lcCse()) << "Could not generate the key";
+        failedToInitialize(account);
         return;
     }
 
     qCInfo(lcCse()) << "Key correctly generated";
     qCInfo(lcCse()) << "Storing keys locally";
 
+    {
+        Bio privKey;
+        if (PEM_write_bio_PrivateKey(privKey, localKeyPair, nullptr, nullptr, 0, nullptr, nullptr) <= 0) {
+            qCInfo(lcCse()) << "Could not read private key from bio.";
+            failedToInitialize(account);
+            return;
+        }
+
+        const auto key = BIO2ByteArray(privKey);
+        _privateKey = key;
+        qCDebug(lcCse) << _privateKey;
+    }
+
     Bio privKey;
     if (PEM_write_bio_PrivateKey(privKey, localKeyPair, nullptr, nullptr, 0, nullptr, nullptr) <= 0) {
         qCInfo(lcCse()) << "Could not read private key from bio.";
+        failedToInitialize(account);
         return;
     }
-    QByteArray key = BIO2ByteArray(privKey);
-    //_privateKey = QSslKey(key, QSsl::Rsa, QSsl::Pem, QSsl::PrivateKey);
-    _privateKey = key;
+
+    auto privateKey = PKey::readPrivateKey(privKey);
 
     qCInfo(lcCse()) << "Keys generated correctly, sending to server.";
-    generateCSR(account, std::move(localKeyPair));
+    auto csrOutput = generateCSR(account, std::move(localKeyPair), std::move(privateKey));
+    writeMnemonic(account, [account, keyPair = std::move(csrOutput.second), output = std::move(csrOutput.first), this]() mutable -> void {
+        writeKeyPair(account, std::move(keyPair), output);
+    });
 }
 
-void ClientSideEncryption::generateCSR(const AccountPtr &account, PKey keyPair)
+std::pair<QByteArray, ClientSideEncryption::PKey> ClientSideEncryption::generateCSR(AccountPtr account,
+                                                                                    PKey keyPair,
+                                                                                    PKey privateKey)
 {
+    auto result = QByteArray{};
```

Comment:
since it's known to always have a `QByteArray` type, plus, you can avoid `copy-assignment`, why not just go with `QByteArray result;`, which will be `{}` by default, anyways?

Reply:
I would expect this to be optimized by the compiler (at least would be with c++20)

Reply:
to be honest, I would like to keep auto `varName` as this then does not stand out of the other variable declarations